setup.py

```diff
@@ -1,22 +1,27 @@
+"""Setup module for the myco-api-client."""
+
 import os
-from setuptools import setup, find_packages
+
+from setuptools import find_packages, setup
+
 from myco_api_client import __version__
 
-target_branch = os.environ.get("BRANCH", "master")
+TARGET_BRANCH = os.environ.get("TARGET_BRANCH", "master")
```

Comment:
This will break the CI, please either revert this (I would actually recommend that) or change the CI scripts in the d3a-web repo.

Reply:
You're right, I missed that.

d3a-web uses `TARGET_BRANCH` to change the tests' target branch: https://github.com/gridsingularity/d3a-web/blob/83d9487b6c075d91a7c65bae9eff97df0d7d8473/tox.ini#L27 .

The important thing for me would be to unify this across the repos, so that we can use the same command to run tox tests in every repo, e.g.:
```sh
(export TARGET_BRANCH=bug/D3ASIM-3572 && tox -e ci -r)
```
I think that `TARGET_BRANCH` is a better name than `BRANCH`, that's why I opted for it. Since this would involve changing the CI scripts, do you suggest renaming it `BRANCH` in all repos instead?

Reply:
I would actually agree with you and be in favor of renaming it to TARGET_BRANCH for all repos and CI scripts. I would rather leave it out of this PR and open a separate task for this in order to test it in isolation. Also tagging @mclrch to be aware of what we are planning to do.

Reply:
Good, then I'll revert the changes in this PR and open a new ticket for it 👍

Reply:
I think this variable BRANCH is not used at all here
We are using TARGET_BRANCH in d3a-web which is retrieves the target branch from github actions:
https://github.com/gridsingularity/d3a-web/blob/83d9487b6c075d91a7c65bae9eff97df0d7d8473/.github/workflows/d3a-web-ci.yaml#L7

Reply:
Another suggestion, we can pass the TARGET_BRANCH as a input to the workflow, smth like: https://github.com/gridsingularity/d3a-web/blob/83d9487b6c075d91a7c65bae9eff97df0d7d8473/.github/workflows/d3a-web-production.yaml#L5

Reply:
@mclrch @spyrostz I'm trying to follow the `BRANCH` variable across the various repos.

The `BRANCH` variable is used in the docker-compose files of d3a-web: https://github.com/gridsingularity/d3a-web/blob/83d9487b6c075d91a7c65bae9eff97df0d7d8473/docker-compose.yml#L19

This variable should affect the deploy scripts and the Dockerfiles for kubernetes. Therefore it will be used to download a specific branch of d3a: https://github.com/gridsingularity/d3a-web/blob/83d9487b6c075d91a7c65bae9eff97df0d7d8473/compose/simulation/Dockerfile_k8s#L5

Also, if `BRANCH` is passed along to the newly-built d3a environment, then the d3a-interface version and the CI tests should also be affected: https://github.com/gridsingularity/d3a/blob/2ea93c8fc80c41f7edd88088e2c65beaa28f8d28/tox.ini#L79

Is this correct, or did I get lost?

Reply:
For deployment we have static associations between environments <-> branch.
For CI you may want to use more flexible settings due to the dependencies, e.g: d3a-interface branch XYZ is incompatible with d3a master branch so you may want to use the branch XYZ of d3a instead of master.
But to be honest I think would be much easier to handle dependencies using tags-versions/bump versions in requirements than branches.
Using tags-versions you can change d3a-interface for example you can commit/merge/create tag and when d3a is ready for changes you can bump the version of the requirements without breaking the enviroment or coordinate merges between repos.